packages/alfa-dom/test/matches.spec.tsx

```diff
@@ -183,3 +183,21 @@ test("Matches an element against a functional host selector", t => {
   t(matches(host, host, ":host(.foo)", { treeContext: root }));
   t(!matches(host, host, ":host(.bar)", { treeContext: root }));
 });
+
+test("Matches an element against a host-context selector", t => {
+  const host = (
+    <div class="bar">
+      <shadow>
+        <span class="child" />
+      </shadow>
+    </div>
+  );
+  const context = <div class="foo">{host}</div>;
+
+  const root = host.shadowRoot!;
+
+  t(matches(host, context, ":host-context(.foo)", { treeContext: root }));
+  t(matches(host, host, ":host-context(.bar)", { treeContext: root }));
```

Comment:
I'm guessing you mean for this to be `matches(host, context, ...)`?

Reply:
Oh, yeah. That wasn't supposed to slip through.